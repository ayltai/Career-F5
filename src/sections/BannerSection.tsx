import { Place as PlaceIcon, Schedule as ScheduleIcon, WorkOutline as WorkIcon, } from '@mui/icons-material';
import { Box, Container, Stack, Typography, useMediaQuery, } from '@mui/material';
import React, { ReactNode, } from 'react';
import { useTranslation, } from 'react-i18next';

import type { Section, } from '../@types';

export const BannerSection = ({
    title,
    description,
    background,
    foregroundImage,
    callToAction,
} : Section & {
    background?      : string,
    foregroundImage? : string,
    callToAction?    : ReactNode,
}) => {
    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const info = (
        <Stack
            alignItems={desktopMode ? undefined : 'center'}
            direction='column'>
            <Typography
                display='flex'
                alignItems='center'
                color='text.secondary'
                variant='body2'>
                <ScheduleIcon />&nbsp;{t('banner_part_time')}
            </Typography>
            <Typography
                display='flex'
                alignItems='center'
                color='text.secondary'
                variant='body2'>
                <PlaceIcon />&nbsp;{t('banner_remote')}
            </Typography>
            <Typography
                display='flex'
                alignItems='center'
                color='text.secondary'
                variant='body2'>
                <WorkIcon />&nbsp;{t('banner_interview')}
            </Typography>
        </Stack>
    );

    return desktopMode ? (
        <Box
            sx={{
                background,
            }}
            height={512}>
            <Container maxWidth='lg'>
                <Stack
                    paddingLeft={2}
                    direction='row'>
                    <Stack
                        width='60%'
                        direction='column'
                        justifyContent='center'>
                        <Typography
                            gutterBottom
                            variant='h2'
                            fontWeight='bold'>
                            {title}
                        </Typography>
                        <Typography
                            gutterBottom
                            variant='h6'>
                            {description}
                        </Typography>
                        {callToAction && (
                            <Box paddingY={3}>
                                {callToAction}
                            </Box>
                        )}
                        {info}
                    </Stack>
                    <Box
                        width='40%'
                        display='flex'>
                        <Box
                            height={512}
                            marginX='auto'
                            marginRight={0}
                            alignItems='center'>
                            <img
                                style={{
                                    marginTop : 32,
                                }}
                                width={317}
                                height={480}
                                alt={title}
                                src={foregroundImage} />
                        </Box>
                    </Box>
                </Stack>
            </Container>
        </Box>
    ) : (
        <Box sx={{
            background,
        }}>
            <Stack
                padding={2}
                direction='column'
                justifyContent='center'
                textAlign='center'>
                <Typography
                    gutterBottom
                    variant='h2'
                    fontWeight='bold'>
                    {title}
                </Typography>
                <Typography
                    gutterBottom
                    variant='h6'>
                    {description}
                </Typography>
                <Box
                    paddingTop={2}
                    paddingBottom={3}>
                    {callToAction}
                </Box>
                {info}
                <Box
                    height={512}
                    marginX='auto'
                    marginBottom={-2}
                    alignItems='center'>
                    <img
                        style={{
                            marginTop : 32,
                        }}
                        width={317}
                        height={480}
                        alt={title}
                        src={foregroundImage} />
                </Box>
            </Stack>
        </Box>
    );
};
