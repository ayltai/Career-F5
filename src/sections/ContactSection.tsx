import { Box, Container, Stack, Typography, useMediaQuery, } from '@mui/material';
import React, { ReactNode, } from 'react';

import type { Section, } from '../@types';
import { FormGenerator, FormGeneratorProps, } from '../components/FormGenerator';

export const ContactSection = ({
    background,
    category,
    title,
    description,
    definition,
    contactAction,
    onSubmit,
} : Section & FormGeneratorProps & {
    background?    : string,
    category?      : string,
    contactAction? : ReactNode,
}) => {
    const desktopMode = useMediaQuery('(min-width:900px)');

    return (
        <Box
            sx={{
                background,
            }}
            paddingY={8}>
            <Container maxWidth='lg'>
                <Stack
                    padding={desktopMode ? 2 : 1}
                    direction={desktopMode ? 'row' : 'column'}>
                    <Stack
                        width={desktopMode ? '50%' : '100%'}
                        padding={desktopMode ? 2 : 0}
                        direction='column'>
                        <Typography
                            gutterBottom
                            color='secondary'
                            variant='subtitle1'
                            fontWeight='bold'>
                            {category}
                        </Typography>
                        <Typography
                            gutterBottom
                            variant='h3'
                            fontWeight='bold'>
                            {title}
                        </Typography>
                        <Typography variant='h6'>
                            {description}
                        </Typography>
                        {contactAction && (
                            <Box paddingY={4}>
                                {contactAction}
                            </Box>
                        )}
                    </Stack>
                    <Box
                        width={desktopMode ? '50%' : '100%'}
                        padding={desktopMode ? 2 : 0}>
                        <FormGenerator
                            definition={definition}
                            onSubmit={onSubmit} />
                    </Box>
                </Stack>
            </Container>
        </Box>
    );
};
