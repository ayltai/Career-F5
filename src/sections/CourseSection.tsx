import { Box, Card, CardActions, CardContent, Chip, Container, Grid, Stack, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

import type { Course, Section, } from '../@types';

export const CourseSection = ({
    backgroundColor,
    category,
    title,
    courses,
} : Section & {
    backgroundColor? : string,
    courses          : Course[],
}) => {
    const { t, } = useTranslation();

    return (
        <Box
            sx={{
                backgroundColor,
            }}
            paddingTop={8}
            paddingBottom={4}>
            <Container maxWidth='lg'>
                <Stack
                    direction='column'
                    textAlign='center'>
                    <Typography
                        gutterBottom
                        color='secondary'
                        variant='subtitle1'
                        fontWeight='bold'>
                        {category}
                    </Typography>
                    <Container maxWidth='md'>
                        <Typography
                            variant='h3'
                            fontWeight='bold'>
                            {title}
                        </Typography>
                    </Container>
                    <Box marginY={8}>
                        <Grid
                            container
                            rowSpacing={{
                                xs : 3,
                                sm : 4,
                                md : 5,
                            }}
                            columnSpacing={{
                                xs : 1,
                                sm : 2,
                                md : 3,
                            }}
                            justifyContent='space-around'>
                            {courses.map(course => (
                                <Grid
                                    key={course.route || course.text}
                                    sx={{
                                        display : 'flex',
                                    }}
                                    item
                                    xs={12}
                                    md={3}>
                                    <Card
                                        sx={{
                                            display       : 'flex',
                                            flexDirection : 'column',
                                            ':hover'      : {
                                                boxShadow : 8,
                                            },
                                        }}
                                        elevation={2}>
                                        <CardContent sx={{
                                            padding       : 2,
                                            display       : 'flex',
                                            flexDirection : 'column',
                                            flexGrow      : 1,
                                        }}>
                                            <Typography
                                                variant='subtitle1'
                                                color='text.secondary'>
                                                {t(course.category)}
                                            </Typography>
                                            <Typography variant='h5'>
                                                {course.text && t(course.text)}
                                            </Typography>
                                        </CardContent>
                                        {course.tags && (
                                            <CardActions sx={{
                                                padding : 2,
                                            }}>
                                                <Stack
                                                    useFlexGap
                                                    spacing={1}
                                                    direction='row'
                                                    flexWrap='wrap'
                                                    justifyContent='flex-start'>
                                                    {course.tags.map(tag => (
                                                        <Chip
                                                            key={tag}
                                                            size='small'
                                                            label={tag} />
                                                    ))}
                                                </Stack>
                                            </CardActions>
                                        )}
                                    </Card>
                                </Grid>
                            ))}
                        </Grid>
                    </Box>
                </Stack>
            </Container>
        </Box>
    );
};
