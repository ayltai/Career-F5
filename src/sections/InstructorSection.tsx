import { OpenInNew as OpenInNewIcon, } from '@mui/icons-material';
import { Avatar, Box, Container, Grid, Link, Stack, Typography, useMediaQuery, } from '@mui/material';
import mixpanel from 'mixpanel-browser';
import React, { Fragment, } from 'react';
import { useTranslation, } from 'react-i18next';

import type { Section, } from '../@types';
import { experiences, } from '../data';

export const InstructorSection = ({
    category,
    title,
} : Section) => {
    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const descriptions : string[] = t('instructor_descriptions', {
        returnObjects : true,
    });

    const handleClick = () => {
        if (process.env.REACT_APP_MIXPANEL_TOKEN) mixpanel.track('Referral/LinkedIn');

        return true;
    };

    const heading = (
        <Fragment>
            <Typography
                gutterBottom
                marginTop={4}
                color='secondary'
                variant='subtitle1'
                fontWeight='bold'>
                {category}
            </Typography>
            <Typography
                variant='h3'
                fontWeight='bold'>
                {title}
            </Typography>
        </Fragment>
    );

    const avatar = (
        <Avatar
            sx={{
                width      : desktopMode ? 480 : '100%',
                height     : desktopMode ? 480 : '100%',
                margin     : desktopMode ? 8 : undefined,
                marginLeft : desktopMode ? 4 : undefined,
                marginTop  : desktopMode ? undefined : 4,
            }}
            alt={t<string>('instructor_header')}
            src='/images/Profile-Business-Small.webp' />
    );

    const logos = (
        <Grid
            container
            marginBottom={4}
            rowSpacing={{
                xs : 3,
                sm : 4,
                md : 5,
            }}
            columnSpacing={{
                xs : 1,
                sm : 2,
                md : 3,
            }}
            justifyContent='space-around'>
            {experiences.map(experience => (
                <Grid
                    key={experience.key}
                    container
                    item
                    xs={6}
                    alignItems='center'
                    textAlign='center'>
                    <Box sx={{
                        marginX : 'auto',
                    }}>
                        <img
                            style={{
                                width     : desktopMode ? 180 : 120,
                                objectFit : desktopMode ? 'contain' : 'scale-down',
                            }}
                            alt={t<string>(experience.key)}
                            src={experience.path} />
                    </Box>
                </Grid>
            ))}
        </Grid>
    );

    const description = (
        <Typography
            marginTop={4}
            variant='body1'>
            <Link
                underline='hover'
                href='https://www.linkedin.com/in/ayltai/'
                target='_blank'
                onClick={handleClick}>
                {t('instructor_name')}
                <OpenInNewIcon fontSize='small' />
            </Link>
            {descriptions.map(desc => (
                <Fragment key={desc}>
                    {desc}<br /><br />
                </Fragment>
            ))}
        </Typography>
    );

    return (
        <Container maxWidth='lg'>
            {desktopMode && (
                <Grid container>
                    <Grid
                        item
                        xs={6}>
                        <Box
                            width='100%'
                            flexDirection='column'>
                            {avatar}
                            {logos}
                        </Box>
                    </Grid>
                    <Grid
                        item
                        paddingX={8}
                        paddingTop={2}
                        paddingBottom={8}
                        xs={6}
                        display='flex'
                        flexDirection='column'
                        justifyContent='center'>
                        {heading}
                        {description}
                    </Grid>
                </Grid>
            )}
            {!desktopMode && (
                <Stack
                    direction='column'
                    justifyContent='center'
                    textAlign='center'>
                    {heading}
                    {avatar}
                    {description}
                    {logos}
                </Stack>
            )}
        </Container>
    );
};
