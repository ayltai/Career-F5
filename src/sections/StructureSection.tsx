import { Box, Card, CardContent, Chip, Container, Divider, Stack, Step, StepButton, StepContent, Stepper, Typography, useMediaQuery, } from '@mui/material';
import React, { Fragment, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import type { Section, } from '../@types';

export const StructureSection = ({
    backgroundColor,
    category,
    steps,
    titles,
    descriptions,
    learnings,
} : Section & {
    backgroundColor? : string,
    steps            : string[],
    titles           : string[],
    descriptions     : string[],
    learnings?       : string[][],
}) => {
    const [ currentStep, setCurrentStep, ] = useState(0);

    const { t, } = useTranslation();

    const desktopMode = useMediaQuery('(min-width:900px)');

    const handleClick = (step : number) => () => setCurrentStep(step);

    const content = (
        <Fragment>
            <Typography
                gutterBottom
                variant={desktopMode ? 'h4' : 'h5'}>
                {titles[currentStep]}
            </Typography>
            <Typography
                gutterBottom
                variant='body2'>
                {descriptions[currentStep]}
            </Typography>
            {learnings && learnings[currentStep] && learnings[currentStep].length > 0 && (
                <Fragment>
                    <Divider sx={{
                        marginY : 2,
                    }} />
                    <Typography
                        gutterBottom
                        marginBottom={desktopMode ? undefined : 2}
                        color='text.secondary'
                        variant={desktopMode ? 'subtitle1' : 'subtitle2'}>
                        {t('structure_learning')}
                    </Typography>
                    <Stack
                        useFlexGap
                        spacing={1}
                        direction='row'
                        flexWrap='wrap'
                        justifyContent='flex-start'>
                        {learnings[currentStep].map(learning => (
                            <Chip
                                key={learning}
                                size='small'
                                label={learning} />
                        ))}
                    </Stack>
                </Fragment>
            )}
            <Box height={40} />
        </Fragment>
    );

    return (
        <Box
            sx={{
                backgroundColor,
            }}
            paddingY={8}>
            <Container maxWidth='lg'>
                <Box
                    display='flex'
                    alignItems='center'>
                    <Typography
                        marginX='auto'
                        marginBottom={4}
                        gutterBottom
                        color='secondary'
                        variant='subtitle1'
                        fontWeight='bold'>
                        {category}
                    </Typography>
                </Box>
                <Stepper
                    alternativeLabel={desktopMode}
                    nonLinear
                    activeStep={currentStep}
                    orientation={desktopMode ? 'horizontal' : 'vertical'}>
                    {steps.map((step, index) => (
                        <Step
                            key={step}
                            completed={currentStep > index}>
                            <StepButton
                                color='text.secondary'
                                onClick={handleClick(index)}>
                                {step}
                            </StepButton>
                            {!desktopMode && (
                                <StepContent>
                                    {content}
                                </StepContent>
                            )}
                        </Step>
                    ))}
                </Stepper>
                {desktopMode && (<Box marginTop={4}>
                    <Card>
                        <CardContent sx={{
                            padding : 8,
                        }}>
                            {content}
                        </CardContent>
                    </Card>
                </Box>)}
            </Container>
        </Box>
    );
};
