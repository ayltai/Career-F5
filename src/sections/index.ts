export { BannerSection, } from './BannerSection';
export { ContactSection, } from './ContactSection';
export { CourseSection, } from './CourseSection';
export { InstructorSection, } from './InstructorSection';
export { OverviewSection, } from './OverviewSection';
export { PaymentSection, } from './PaymentSection';
export { ProspectSection, } from './ProspectSection';
export { StructureSection, } from './StructureSection';
