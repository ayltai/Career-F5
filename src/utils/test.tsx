import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
