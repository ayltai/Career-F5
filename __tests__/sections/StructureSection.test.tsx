import React from 'react';

import { fireEvent, render, waitFor, } from '../../src/utils/test';
import { StructureSection, } from '../../src/sections/StructureSection';

describe('StructureSection', () => {
    it('renders correctly', () => expect(render(
        <StructureSection
            backgroundColor='backgroundColor'
            category='category'
            steps={[
                'step_1',
                'step_2',
            ]}
            titles={[
                'title_1',
                'title_2',
            ]}
            descriptions={[
                'description_1',
                'description_2',
            ]}
            learnings={[
                [
                    'tag_1',
                    'tag_2',
                ],
                [
                    'tag_3',
                    'tag_4',
                ],
            ]} />
    )).toMatchSnapshot());

    it('renders correctly with steps', async () => {
        const { getAllByRole, getByText, queryByText, } = render(
            <StructureSection
                backgroundColor='backgroundColor'
                category='category'
                steps={[
                    'step_1',
                    'step_2',
                ]}
                titles={[
                    'title_1',
                    'title_2',
                ]}
                descriptions={[
                    'description_1',
                    'description_2',
                ]}
                learnings={[
                    [
                        'tag_1',
                        'tag_2',
                    ],
                    [
                        'tag_3',
                        'tag_4',
                    ],
                ]} />
        );

        expect(getByText('title_1')).toBeInTheDocument();
        expect(queryByText('title_2')).not.toBeInTheDocument();

        fireEvent.click(getAllByRole('button')[1]);

        await waitFor(() => {
            expect(getByText('title_2')).toBeInTheDocument();
            expect(queryByText('title_1')).not.toBeInTheDocument();
        });
    });
});
