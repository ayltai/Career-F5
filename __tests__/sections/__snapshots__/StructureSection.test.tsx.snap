// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructureSection renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-12vwxo4"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-70qvj9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-8timqt-MuiTypography-root"
            >
              category
            </h6>
          </div>
          <div
            class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
          >
            <div
              class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
            >
              <button
                aria-current="step"
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                color="text.secondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-kkz1hq-MuiSvgIcon-root-MuiStepIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="12"
                      />
                      <text
                        class="MuiStepIcon-text css-opt7yd-MuiStepIcon-text"
                        dominant-baseline="central"
                        text-anchor="middle"
                        x="12"
                        y="12"
                      >
                        1
                      </text>
                    </svg>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label Mui-active css-25v9v6-MuiStepLabel-label"
                    >
                      step_1
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiStepContent-root css-14yr603-MuiStepContent-root"
              >
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiStepContent-transition MuiCollapse-entered css-1rw7nzj-MuiCollapse-root-MuiStepContent-transition"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <h5
                        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1jd4bdu-MuiTypography-root"
                      >
                        title_1
                      </h5>
                      <p
                        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-2m1ky5-MuiTypography-root"
                      >
                        description_1
                      </p>
                      <hr
                        class="MuiDivider-root MuiDivider-fullWidth css-ctpjxq-MuiDivider-root"
                      />
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1hb2odc-MuiTypography-root"
                      >
                        structure_learning
                      </h6>
                      <div
                        class="MuiStack-root css-1bzkvkb-MuiStack-root"
                      >
                        <div
                          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-j82tey-MuiChip-root"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                          >
                            tag_1
                          </span>
                        </div>
                        <div
                          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-j82tey-MuiChip-root"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                          >
                            tag_2
                          </span>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-u1jr60"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-vertical css-1pe7n21-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineVertical css-8t49rw-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
            >
              <button
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                color="text.secondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-kkz1hq-MuiSvgIcon-root-MuiStepIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="12"
                      />
                      <text
                        class="MuiStepIcon-text css-opt7yd-MuiStepIcon-text"
                        dominant-baseline="central"
                        text-anchor="middle"
                        x="12"
                        y="12"
                      >
                        2
                      </text>
                    </svg>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label css-25v9v6-MuiStepLabel-label"
                    >
                      step_2
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiStepContent-root MuiStepContent-last css-1mz1l2x-MuiStepContent-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-12vwxo4"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-8timqt-MuiTypography-root"
          >
            category
          </h6>
        </div>
        <div
          class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
        >
          <div
            class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
          >
            <button
              aria-current="step"
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              color="text.secondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-kkz1hq-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-opt7yd-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      1
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-25v9v6-MuiStepLabel-label"
                  >
                    step_1
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiStepContent-root css-14yr603-MuiStepContent-root"
            >
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiStepContent-transition MuiCollapse-entered css-1rw7nzj-MuiCollapse-root-MuiStepContent-transition"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1jd4bdu-MuiTypography-root"
                    >
                      title_1
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-2m1ky5-MuiTypography-root"
                    >
                      description_1
                    </p>
                    <hr
                      class="MuiDivider-root MuiDivider-fullWidth css-ctpjxq-MuiDivider-root"
                    />
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1hb2odc-MuiTypography-root"
                    >
                      structure_learning
                    </h6>
                    <div
                      class="MuiStack-root css-1bzkvkb-MuiStack-root"
                    >
                      <div
                        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-j82tey-MuiChip-root"
                      >
                        <span
                          class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                        >
                          tag_1
                        </span>
                      </div>
                      <div
                        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-j82tey-MuiChip-root"
                      >
                        <span
                          class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                        >
                          tag_2
                        </span>
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-u1jr60"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-vertical css-1pe7n21-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineVertical css-8t49rw-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
          >
            <button
              class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
              color="text.secondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-kkz1hq-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-opt7yd-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      2
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label css-25v9v6-MuiStepLabel-label"
                  >
                    step_2
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiStepContent-root MuiStepContent-last css-1mz1l2x-MuiStepContent-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
